src/GafferScene/ShaderPlug.cpp

```diff
@@ -139,12 +119,43 @@ bool ShaderPlug::acceptsInput( const Gaffer::Plug *input ) const
 
 IECore::MurmurHash ShaderPlug::attributesHash() const
 {
-	const Shader *s = shader( this );
-	return s ? s->attributesHash() : MurmurHash();
+	const Gaffer::Plug *p = shaderOutPlug();
+
+	IECore::MurmurHash h;
+	if (p)
+	{
+		const GafferScene::Shader *s = runTimeCast<const GafferScene::Shader>( p->node() );
```

Comment:
The `shaderOutPlug()` name implied to me that the function returned the output plug of a shader. And both places that call the function appear to want exactly that. But the function actually returns any old plug without checking it is a shader output, and then both call sites have to perform that check themselves. I was suggesting you move the "is this plug the child of a Shader node?" check to `shaderOutPlug()`, and return NULL if not. But I realise now that you need to use the Shader node anyway, so there's no real benefit, so let's scratch this...